ParseTree/ParseTree/DataHandler.cs

```diff
@@ -0,0 +1,150 @@
+﻿// <copyright file="DataHandler.cs" company="Roman Levashev">
+// Copyright (c) Roman Levashev. All rights reserved.
+// Licensed under the MIT License.
+// </copyright>
+
+namespace ParseTree;
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using ParseTree.Nodes;
+
+/// <summary>
+/// Provides functionality for parsing and evaluating expressions using parse trees.
+/// </summary>
+public static class DataHandler
+{
+    /// <summary>
+    /// Parses a string input into a parse tree structure.
+    /// </summary>
+    /// <param name="input">The string expression to parse.</param>
+    /// <returns>The root <see cref="Node"/> of the constructed parse tree.</returns>
+    public static Node Parse(string input)
+    {
+        var tokens = Tokenize(input);
+        var enumerator = tokens.GetEnumerator();
+        if (!enumerator.MoveNext())
+        {
+            throw new FormatException("Empty expression");
+        }
+
+        var result = ParseExpression(enumerator);
+
+        if (result.ResultEnumerator.Current != null)
+        {
+            throw new FormatException($"Unexpected extra tokens: '{enumerator.Current}'");
+        }
+
+        return result.Root;
+    }
+
+    /// <summary>
+    /// Evaluates and prints both the parse tree structure and the expression result.
+    /// </summary>
+    /// /// <param name="expression">The mathematical expression to evaluate.</param>
+    public static void EvaluateAndPrintExpression(string expression)
+    {
+        var root = Parse(expression);
+        Console.Write("Parse Tree: ");
+        root.Print();
+        Console.WriteLine();
+        Console.WriteLine($"Expression result: {root.Evaluate()}");
+    }
+
+    private static (Node Root, IEnumerator<string> ResultEnumerator) ParseExpression(IEnumerator<string> enumerator)
+    {
+        var token = enumerator.Current;
+
+        if (int.TryParse(token, out var value))
+        {
+            enumerator.MoveNext();
+            return (new NumberNode(value), enumerator);
+        }
+        else if (token == '('.ToString())
```

Comment:
```suggestion
        else if (token == "(")
```
Странная запись просто. Кажется, мы и так знаем, чему равно '('.ToString()